clippy_lints/src/booleans.rs

```diff
@@ -473,6 +496,12 @@ impl<'a, 'tcx> Visitor<'tcx> for NonminimalBoolVisitor<'a, 'tcx> {
                     self.bool_expr(e);
                 },
                 ExprKind::Unary(UnOp::Not, inner) => {
+                    if let ExprKind::Unary(UnOp::Not, ex) = inner.kind {
+                        let ty = self.cx.typeck_results().expr_ty(ex);
+                        if is_impl_not_trait_with_bool_out(self.cx, ty) {
+                            return;
+                        }
```

Comment:
Out of curiosity, why not just check if `ex` is a `bool`?

Reply:
Thanks you for your feedback. I try this

Reply:
Are you some method to suggest to me please? When I try `cx.typeck_results().expr_ty(..)` on struct, it returns struct name, not bool type